src/chart/generateCategoricalChart.tsx

```diff
@@ -1828,7 +1828,10 @@ export const generateCategoricalChart = ({
       return (
         <CartesianAxis
           {...(axisOptions as any)}
-          className={`recharts-${axisOptions.axisType} ${axisOptions.axisType}`}
+          className={classNames(
+            `recharts-${axisOptions.axisType} ${axisOptions.axisType}`,
+            (axisOptions as any).className,
```

Comment:
would love for less things to be cast to `any` rather than more. These types are a mess but would prefer `className` added explicitly to `axisOptions` rather than cast here